language/move-vm/types/src/values/values_impl.rs

```diff
@@ -263,6 +271,33 @@ impl Value {
                 (SignatureToken::Signer, Container::Resource(v)) => {
                     v.len() == 1 && matches!(&v[0], ValueImpl::Address(_))
                 }
+                _ => true,
+            },
+            _ => false,
+        }
+    }
+
+    pub fn is_constant_or_sender_ref(&self) -> bool {
```

Comment:
this is just a check that the type is a constant or a sender reference. In any case this is the check from the Move Adapter that an argument is from a limited set of types

Reply:
nit, `s/sender/signer/` ?